src/prefect/cli/deploy.py

```diff
@@ -571,41 +571,46 @@ async def _run_single_deploy(
 
     work_pool = await client.read_work_pool(deploy_config["work_pool"]["name"])
 
-    if is_interactive() and not docker_build_step_exists and not build_step_set_to_null:
-        docker_based_infrastructure = "image" in work_pool.base_job_template.get(
-            "variables", {}
-        ).get("properties", {})
-        if docker_based_infrastructure:
-            build_docker_image_step = await prompt_build_custom_docker_image(
-                app.console, deploy_config
+    image_is_configurable = (
+        "image"
+        in work_pool.base_job_template.get("variables", {}).get("properties", {})
+        # managed work pools cannot be configured with a custom image
+        and not work_pool.type.endswith(":managed")
```

Comment:
Might be worthwhile to key off of the image variable instead of the work pool type. Managed work pools only accept a set of images from an enum, so we should only prompt for building a custom image if the image variable accepts a string.

Reply:
good call! updated in [4d0116d](https://github.com/PrefectHQ/prefect/pull/15484/commits/4d0116d42e7f47ce8a80c92015d2998998a84e95)